src/OrleansSQLUtils/Storage/RelationalStorage.cs

```diff
@@ -195,35 +201,55 @@ private RelationalStorage(string invariantName, string connectionString)
         }
 
 
-        private static async Task<Tuple<IEnumerable<TResult>, int>> ExecuteReaderAsync<TResult>(DbCommand command, Func<IDataRecord, int, TResult> selector)
+        private async Task<Tuple<IEnumerable<TResult>, int>> ExecuteReaderAsync<TResult>(DbCommand command, Func<IDataRecord, int, CancellationToken, Task<TResult>> selector, CancellationToken cancellationToken, CommandBehavior commandBehavior)
         {
-            using(var reader = await command.ExecuteReaderAsync().ConfigureAwait(continueOnCapturedContext: false))
-            {
-                return await SelectAsync(reader, selector).ConfigureAwait(false);
+            using(var reader = await command.ExecuteReaderAsync(commandBehavior, cancellationToken).ConfigureAwait(continueOnCapturedContext: false))
+            {                                                                
```

Comment:
Maybe a note that not al ADO.NET providers support cancellation is in order, in addition to the function call.